static/src/javascripts/projects/common/modules/video/video-container.js

```diff
@@ -0,0 +1,268 @@
+// @flow
+import bean from 'bean';
+import fastdom from 'lib/fastdom-promise';
+import $ from 'lib/$';
+import ElementInview from 'lib/element-inview';
+import videojs from 'bootstraps/enhanced/media/video-player';
+import youtube from 'common/modules/atoms/youtube';
+import detect from 'lib/detect';
+
+type State = {
+    position: number,
+    length: number,
+    videoWidth: number,
+    container: Element,
+};
+
+type Action = {
+    type: string,
+};
+
+const updateYouTubeVideo = (currentItem: ?Element) => {
+    if (currentItem != null) {
+        const youTubeAtom = currentItem.querySelector('.youtube-media-atom');
+        if (youTubeAtom) {
+            return youtube.onVideoContainerNavigation(
+                youTubeAtom.getAttribute('data-unique-atom-id')
+            );
+        }
+    }
+};
+
+const getPositionState = (position: number, length: number) => ({
+    position,
+    atStart: position === 0,
+    atEnd: position >= length,
+});
+
+const reducers = {
+    NEXT: function next(previousState: State) {
+        const position = previousState.position >= previousState.length
+            ? previousState.position
+            : previousState.position + 1;
+
+        updateYouTubeVideo(
+            document.querySelector(`.js-video-playlist-item-${position - 1}`)
+        );
+        return Object.assign(
+            {},
+            previousState,
+            getPositionState(position, previousState.length)
+        );
+    },
+
+    PREV: function prev(previousState: State) {
+        const position = previousState.position <= 0
+            ? 0
+            : previousState.position - 1;
+        updateYouTubeVideo(
+            document.querySelector(`.js-video-playlist-item-${position + 1}`)
+        );
+        return Object.assign(
+            {},
+            previousState,
+            getPositionState(position, previousState.length)
+        );
+    },
+
+    INIT: function init(previousState: State) {
+        const makeYouTubeNonPlayableAtSmallBreakpoint = state => {
+            if (
+                detect.isBreakpoint({
+                    max: 'desktop',
+                })
+            ) {
+                const youTubeIframes = state.container.querySelectorAll(
+                    '.youtube-media-atom iframe'
+                );
+                youTubeIframes.forEach(el => {
```

Comment:
I think this would break, wouldn't it? `youTubeIframes` is a `NodeList`, where you can't use `.forEach`, so you would have to do `[...youTubeIframes].forEach( ... )`.